src/MemoryRouter.tsx

```diff
@@ -48,8 +55,10 @@ export abstract class BaseRouter implements NextRouter {
   basePath = "";
   isFallback = false;
   events = new EventEmitter();
+  locale: string | undefined = undefined;
+  locales = [];
```

Comment:
I don't see this being used anywhere ... can we add a unit test for it? Should we be able to just access it for tests, like `mockRouter.locales.push('aa', 'bb', 'cc')`?

Reply:
technically I need to so I can setup the test for this component:

```ts
import React, { FunctionComponent, useCallback } from 'react'
import { useRouter } from 'next/router'
import { useTranslation } from 'react-i18next'
import css from './languageSwitch.module.scss'

export const LanguageSwitch: FunctionComponent = () => {
  const router = useRouter()
  const { locale, locales } = useRouter()
  const { t } = useTranslation()

  const _change = useCallback((event) => {
    router.push(router.route, undefined, { locale: event.target.value })
  }, [])

  return (
    <select value={locale} onChange={_change} className={css.select}>
      {locales.map((l) => (
        <option key={l} value={l}>
          {t(`languages.${l}`)}
        </option>
      ))}
    </select>
  )
}
```

without this 2 properties the `useRouter` mock can not work.
It would make very few sense to access it in a test, since it would just test the mock implementation of `push`

Reply:
Well, setting `mockRouter.locales` would be helpful for testing this component ... like:
```
mockRouter.locales = [ 'one', 'two' ]
render(<LanguageSwitch />);
screen.getByValue('one').click();
expect(mockRouter.locale).toEqual('one');
```

Reply:
Let's add just a simple unit test, like:
```
it('should support the locales property', () => {
  expect(mockRouter.locales).toEqual([ ]);
});
```

Reply:
done, and I added an extra test for both on the router to test the feature on the hook